alacritty/src/display.rs

```diff
@@ -192,19 +218,18 @@ impl Display {
         // Update OpenGL projection
         renderer.resize(&size_info);
 
-        // Clear screen
+        // Call `clear` before showing the window, to make sure the surface is initialized.
         let background_color = config.colors.primary.background;
         renderer.with_api(&config, &size_info, |api| {
             api.clear(background_color);
         });
 
-        // We should call `clear` when window is offscreen, so when `window.show()` happens it
-        // would be with background color instead of uninitialized surface.
         #[cfg(not(any(target_os = "macos", windows)))]
         {
-            // On Wayland we can safely ignore this call, since the window isn't visible until you
-            // actually draw something into it.
+            // Setup and perform platform specific helpers
             if event_loop.is_x11() {
+                // On Wayland we can safely ignore this call, since the window isn't visible until
+                // you actually draw something into it.
```

Comment:
Seems like due to the other changes the `else` branch was removed? In that case I'd move it in front of the `is_x11` again since that's a direct x11/Wayland reference.